tests/migration_test.go

```diff
@@ -1019,28 +1053,243 @@ var _ = Describe("[rfe_id:393][crit:high[vendor:cnv-qe@redhat.com][level:system]
+
+					labelKey := "testkey"
+					labels := map[string]string{
+						labelKey: "",
+					}
+
+					// give an affinity rule to ensure the vmi's get placed on the same node.
+					affinityRule := &k8sv1.Affinity{
+						PodAffinity: &k8sv1.PodAffinity{
+							PreferredDuringSchedulingIgnoredDuringExecution: []k8sv1.WeightedPodAffinityTerm{
+								{
+									Weight: int32(1),
+									PodAffinityTerm: k8sv1.PodAffinityTerm{
+										LabelSelector: &metav1.LabelSelector{
+											MatchExpressions: []metav1.LabelSelectorRequirement{
+												{
+													Key:      labelKey,
+													Operator: metav1.LabelSelectorOpIn,
+													Values:   []string{string("")}},
+											},
+										},
+										TopologyKey: "kubernetes.io/hostname",
+									},
+								},
+							},
+						},
+					}
+
+					vmi_evict1.Labels = labels
+					vmi_evict2.Labels = labels
+					vmi_noevict.Labels = labels
+
+					vmi_evict1.Spec.Affinity = affinityRule
+					vmi_evict2.Spec.Affinity = affinityRule
+					vmi_noevict.Spec.Affinity = affinityRule
+
+					By("Starting the VirtualMachineInstance with eviction set to live migration")
+					vm_evict1 := tests.NewRandomVirtualMachine(vmi_evict1, false)
+					vm_evict2 := tests.NewRandomVirtualMachine(vmi_evict2, false)
+					vm_noevict := tests.NewRandomVirtualMachine(vmi_noevict, false)
+
+					// post VMs
+					vm_evict1, err = virtClient.VirtualMachine(tests.NamespaceTestDefault).Create(vm_evict1)
+					Expect(err).ToNot(HaveOccurred())
+					vm_evict2, err = virtClient.VirtualMachine(tests.NamespaceTestDefault).Create(vm_evict2)
+					Expect(err).ToNot(HaveOccurred())
+					vm_noevict, err = virtClient.VirtualMachine(tests.NamespaceTestDefault).Create(vm_noevict)
+					Expect(err).ToNot(HaveOccurred())
+
+					// Start VMs
+					vm_evict1 = tests.StartVirtualMachine(vm_evict1)
+					vm_evict2 = tests.StartVirtualMachine(vm_evict2)
+					vm_noevict = tests.StartVirtualMachine(vm_noevict)
+
+					// Get VMIs
+					vmi_evict1, err = virtClient.VirtualMachineInstance(vmi_evict1.Namespace).Get(vmi_evict1.Name, &metav1.GetOptions{})
+					vmi_evict2, err = virtClient.VirtualMachineInstance(vmi_evict1.Namespace).Get(vmi_evict2.Name, &metav1.GetOptions{})
+					vmi_noevict, err = virtClient.VirtualMachineInstance(vmi_evict1.Namespace).Get(vmi_noevict.Name, &metav1.GetOptions{})
+
+					By("Verifying all VMIs are collcated on the same node")
+					Expect(vmi_evict1.Status.NodeName).To(Equal(vmi_evict2.Status.NodeName))
+					Expect(vmi_evict1.Status.NodeName).To(Equal(vmi_noevict.Status.NodeName))
 
+					// Taint Node.
+					By("Tainting node with kubevirt.io/drain=NoSchedule")
+					node := vmi_evict1.Status.NodeName
+					tests.Taint(node, "kubevirt.io/drain", k8sv1.TaintEffectNoSchedule)
+
+					// Drain Node using cli client
+					By("Draining using kubectl drain")
```

Comment:
I'm not saying the test is wrong, just wanted to double check intent here. It looks like the VM's are started and the the kubectl drain operation is immediately invoked. Would it make sense to wait at least until some subset of the 3 VMI's is confirmed running?

Reply:
StartVirtualMachine() ensures the resulting VMI is running before it returns.